core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -6237,6 +6241,33 @@ private QueryModel wrapWithSelectModel(QueryModel model, IntList insetColumnInde
         return _model;
     }
 
+    private void addColumnToSelectModel(QueryModel model, IntList insetColumnIndexes, ObjList<QueryColumn> insertColumnAliases, CharSequence timestampAlias) {
+        ObjList<QueryColumn> origins = new ObjList<>(model.getBottomUpColumns());
```

Comment:
Can we avoid creating a new list here? If you add methods to QueryModel to insert column by index you can rewrite the code in something like 

```Java
        int src2ColumnCount = insetColumnIndexes.size();
        for (int k = 0; k < src2ColumnCount; k++) {
            int insertIndex = insetColumnIndexes.getQuick(k) + k;
            QueryColumn column = insertColumnAliases.get(k);
            if (column.getAst().type == LITERAL) {
                model.insertBottomUpColumnIfNotExits(insertIndex, nextColumn(column.getAlias(), timestampAlias), false);
            } else {
                model.insertBottomUpColumnIfNotExits(insertIndex, column, false);
            }
        }
```

Reply:
it would be good to quantify inefficient. We are trying to avoid extra garbage allocations. Lets find a solution.